### PR TITLE
add the features of deletion & name generation

### PR DESCRIPTION
**Pull request by sivecanta**

### CODE REVIEW

File: lib/main.dart [hunk 1]
```diff
@@ -36,72 +36,91 @@ class _RandomWordsState extends State<RandomWords> {
...
+        child: const Icon(Icons.plus_one),
+        backgroundColor: Colors.green,
+      ),
     );
   }
 
   Widget _buildSuggestions() {
     return ListView.builder(
       padding: const EdgeInsets.all(16.0),
+      itemCount: _suggestions.length,
```
Comment (max4nik) [inline lib/main.dart]: Here you assign length of suggestions to itemCount variable but it is not using anywhere. Remove it or better use it in line 55

File: lib/main.dart [hunk 2]
```diff
@@ -36,72 +36,91 @@ class _RandomWordsState extends State<RandomWords> {
...
+      final tiles = _saved.map((WordPair pair) {
+        return ListTile(
+          title: Text(
+            pair.asPascalCase,
+            style: _biggerFont,
+          ),
+          trailing: const Icon(Icons.delete),
+          onTap: () {
+            if (_saved.contains(pair)) {
+              _saved.remove(pair);
```
Comment (Denys-Doskochynskyi) [inline lib/main.dart]: Maybe more better move .remove  inside setState()

File: lib/main.dart [hunk 3]
```diff
@@ -36,72 +36,91 @@ class _RandomWordsState extends State<RandomWords> {
...
+      final divided = tiles.isNotEmpty
+          ? ListTile.divideTiles(context: context, tiles: tiles).toList()
+          : <Widget>[];
+      return Scaffold(
+        appBar: AppBar(
+          title: const Text('Saved Suggestions'),
+          actions: [
+            IconButton(
+                icon: const Icon(Icons.delete_forever),
+                onPressed: () {
```
Comment (Denys-Doskochynskyi) [inline lib/main.dart:116]: Maybe more better move .clear inside setState()

File: lib/main.dart [hunk 4]
```diff
@@ -36,72 +36,91 @@ class _RandomWordsState extends State<RandomWords> {
         ],
       ),
       body: _buildSuggestions(),
+      floatingActionButton: FloatingActionButton(
+        onPressed: () {
+          _suggestions.addAll(generateWordPairs().take(5));
```
Comment (Denys-Doskochynskyi) [inline lib/main.dart]: Maybe more better move .addAll inside setState()

File: lib/main.dart [hunk 5]
```diff
@@ -36,72 +36,91 @@ class _RandomWordsState extends State<RandomWords> {
...
+          onTap: () {
+            if (_saved.contains(pair)) {
+              _saved.remove(pair);
+              setState(() {}); // refresh page
+              Navigator.pop(context);
+            }
+          },
+        );
+      });
+      final divided = tiles.isNotEmpty
```
Comment (max4nik) [inline lib/main.dart]: Would be better to sepecify what is divided here. Maybe it's better to rename this variable to something like "dividedTiles" or "dividedItems"

**Review by Denys-Doskochynskyi — COMMENTED** [review]
Non  critical correction, merge pls :)

**Review by max4nik — APPROVED** [review]
All changes look good! 
Take a look at the comments I left, they might be helpful for better understanding and generally for keeping your code clean

**Review by Denys-Doskochynskyi — APPROVED** [review]

**Comment by igor-leshkevych** [comment]
@aekrops please add the description (what your feature does + screenshots)

File: lib/main.dart [hunk 6]
```diff
@@ -36,72 +36,93 @@ class _RandomWordsState extends State<RandomWords> {
...
-              : <Widget>[];
-          return Scaffold(
-            appBar: AppBar(
-              title: Text('Saved Suggestions'),
+    return (alreadySaved)
+        ? Container(color: Colors.white)
+        : ListTile(
+            title: Text(
+              pair.asPascalCase,
+              style: _biggerFont,
```
Comment (yaraoprysk) [inline lib/main.dart:69]: I think, in this case, you don`t need to use a special variable for fontSize

**Review by yaraoprysk — APPROVED** [review]
everything is perfect, just merge it:))

**Comment by sivecanta** [comment]
image
Several features were added:
- The button that adds 5 new suggestion words
- When you add the word to saved words, it disappears from the suggestion list
- In the saved list you can delete one word or delete them all by press the right top trash button

**Comment by sivecanta** [comment]
image
Let's save 3 words. We can see that they disappeared

**Comment by sivecanta** [comment]
image
We can see them in the saved list. Let's delete one word.

**Comment by sivecanta** [comment]
image
And press the top right trash button for deleting the rest words

**Comment by sivecanta** [comment]
image

**Comment by sivecanta** [comment]
Also, pressing the green button adding 5 random words to the suggestion set

File: lib/main.dart (hunk 6 above)
Comment (igor-leshkevych) [reply lib/main.dart:69]: if you left a comment with a fix, than probably it's not that *perfect*. so be careful before leaving comments like that in the future

**Comment by igor-leshkevych** [comment]
@yaraoprysk 
+2/2 code review

File: lib/main.dart [hunk 7]
```diff
@@ -36,72 +36,93 @@ class _RandomWordsState extends State<RandomWords> {
...
             ),
-            body: ListView(children: divided),
+            trailing: Icon(
+              alreadySaved ? Icons.favorite : Icons.favorite_border,
+              color: alreadySaved ? Colors.red : null,
```
Comment (igor-leshkevych) [inline lib/main.dart:73]: I wouldn't recommend using `null` for colors

**Review by igor-leshkevych — APPROVED** [review]
@aekrops 

It's better to update PR description and add screenshots there rather than adding them as separate messages in conversations.

+5/5 hello world
+3/3 custom feature + PR

**Pull request merged**